packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -2252,11 +2284,92 @@ export class HealthCheck {
   private constructor(public readonly type: string, public readonly gracePeriod?: Duration) { }
 }
 
+/**
+ * Heath checks base options
+ */
+interface HealthChecksBaseOptions {
+  /**
+   * Specified the time Auto Scaling waits before checking the health status of an EC2 instance that has come into service
+   * and marking it unhealthy due to a failed health check.
+   *
+   * @default Duration.seconds(0)
+   * @see https://docs.aws.amazon.com/autoscaling/ec2/userguide/health-check-grace-period.html
+   */
+  readonly grace?: Duration;
+}
+
+/**
+ * EC2 Heath checks options
+ */
+export interface Ec2HealthChecksOptions extends HealthChecksBaseOptions {
+}
+
+/**
+ * Additional Heath checks options
+ */
+export interface AdditionalHealthChecksOptions extends HealthChecksBaseOptions {
+  /**
+   * One or more health check types other than EC2.
+   */
+  readonly additionalTypes: AdditionalHealthCheckType[];
+}
+
+/**
+ * Health check settings for multiple types
+ */
+export class HealthChecks {
+  /**
+   * Use EC2 for health checks
+   *
+   * @param options EC2 health checks options
+   */
+  public static ec2(options: Ec2HealthChecksOptions = {}): HealthChecks {
+    return new HealthChecks(['EC2'], options.grace);
+  }
+
+  /**
+   * Use additional health checks.
+   * It considers the instance unhealthy if it fails either the EC2 status checks or the additional health checks.
+   *
+   * @param options Additional health checks options
+   */
+  public static addition(options: AdditionalHealthChecksOptions): HealthChecks {
+    return new HealthChecks(options.additionalTypes, options.grace);
+  }
+
+  private constructor(public readonly types: string[], public readonly gracePeriod?: Duration) {
+    if (types.length === 0) {
+      throw new Error('At least one health check type must be specified in \'additionalTypes\' for \'healthChecks\'');
+    }
+  }
+}
+
+/**
+ * @deprecated Use AdditionalHealthCheckType instead
+ */
 enum HealthCheckType {
   EC2 = 'EC2',
   ELB = 'ELB',
 }
 
+/**
+ * Additional Health Check Type
+ */
+export enum AdditionalHealthCheckType {
+  /**
+   * ELB Health Check
+   */
+  ELB = 'ELB',
+  /**
+   * EBS Health Check
+   */
+  EBS = 'EBS',
+  /**
+   * VPC LATTICE Health Check
+   */
+  VPC_LATTICE = 'VPC_LATTICE',
+}
```

Comment:
EC2 had to be excluded because specifying EC2 and other types together would result in a CFn error.

```
Resource handler returned message: "Specifying EC2 in addition to other health check types is not supported. Remove EC2 from HealthCheckType and try again. (Service: AutoScaling, Status Code: 400, Request ID: ...
```